plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -234,6 +240,28 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
             }
             ++current_request->start_block_num;
          }
+
+         auto& block_num = current_request->start_block_num;
+         auto get_blk = [&chain, block_num, block_state]() -> signed_block_ptr {
+            try {
+               if (block_state && block_state->block_num == block_num)
+                  return block_state->block;
+               return chain.fetch_block_by_number(block_num);
+            } catch (...) {
+               return {};
+            }
+         };
+         auto block = get_blk();
+
+         bool fresh_block = block && fc::time_point::now() - block->timestamp < fc::minutes(log_reduction_minutes);
+         if( fresh_block || (result.this_block && result.this_block->block_num % log_reduction_one_per == 0) ) {
+            ilog("pushing result "
```

Comment:
If https://github.com/eosnetworkfoundation/mandel/pull/595 is merged before this, then change this to `fc_ilog`.

Reply:
Merged and changed